examples/jsm/geometries/SDFGeometry.js

```diff
@@ -0,0 +1,319 @@
+        var e0 = cube_edges[ i<<1 ]       //Unpack vertices
+          , e1 = cube_edges[(i<<1)+1]
+          , g0 = grid[e0]                 //Unpack grid values
+          , g1 = grid[e1]
+          , t  = g0 - g1;                 //Compute point of intersection
+        if(Math.abs(t) > 1e-6) {
+          t = g0 / t;
+        } else {
+          continue;
+        }
+        
+        //Interpolate vertices and add up intersections (this can be done without multiplying)
+        for(var j=0, k=1; j<3; ++j, k<<=1) {
+          var a = e0 & k
+            , b = e1 & k;
+          if(a !== b) {
+            v[j] += a ? 1.0 - t : t;
+          } else {
+            v[j] += a ? 1.0 : 0;
+          }
+        }
+      }
+      
+      //Now we just average the edge intersections and add them to coordinate
+      var s = 1.0 / e_count;
+      for(var i=0; i<3; ++i) {
+        v[i] = scale[i] * (x[i] + s * v[i]) + shift[i];
+      }
+      
+      //Add vertex to buffer, store pointer to vertex index in buffer
+      buffer[m] = vertices.length;
+      vertices.push(v);
+      
+      //Now we need to add faces together, to do this we just loop over 3 basis components
+      for(var i=0; i<3; ++i) {
+        //The first three entries of the edge_mask count the crossings along the edge
+        if(!(edge_mask & (1<<i)) ) {
+          continue;
+        }
+        
+        // i = axes we are point along.  iu, iv = orthogonal axes
+        var iu = (i+1)%3
+          , iv = (i+2)%3;
+          
+        //If we are on a boundary, skip it
+        if(x[iu] === 0 || x[iv] === 0) {
+          continue;
+        }
+        
+        //Otherwise, look up adjacent edges in buffer
+        var du = R[iu]
+          , dv = R[iv];
+        
+        //Remember to flip orientation depending on the sign of the corner.
+        if(mask & 1) {
+          faces.push([buffer[m],    buffer[m-du],    buffer[m-dv]]);
+          faces.push([buffer[m-dv], buffer[m-du],    buffer[m-du-dv]]);
+        } else {
+          faces.push([buffer[m],    buffer[m-dv],    buffer[m-du]]);
+          faces.push([buffer[m-du], buffer[m-dv],    buffer[m-du-dv]]);
+        }
+      }
+    }
+  }
+  
+  //All done!  Return the result
+  return { positions: vertices, cells: faces };
+};
+
+import { 
+    BufferAttribute, 
+    BufferGeometry, 
+    FloatType, 
+    Mesh, 
+    OrthographicCamera, 
+    PlaneGeometry, 
+    Scene, 
+    ShaderMaterial, 
+    Vector2, 
+    WebGLRenderer, 
+    WebGLRenderTarget 
+} from "three"
+
+let sdfCompute = `
+    varying vec2 vUv;
+    uniform float tileNum;
+    uniform float bounds;
+
+    [#dist#]
+
+    void main()	{ gl_FragColor=vec4( ( dist( vec3( vUv, tileNum ) * 2.0 * bounds - vec3( bounds ) ) < 0.00001 ) ? 1.0 : 0.0 ); }
+`
+
+let defaultDist = `
+    float dist( vec3 p ){
+        float c = length(p) - 0.5;
+        float comp = min( 1e10, c );
+        return comp; 
+    }
+`
+
+class SDFtoMesh extends BufferGeometry{
```

Comment:
If you want to extend the class from `BufferGeometry`, name it `SDFGeometry`.

Maybe you can also think of a different design. Something like `SDFGenerator`. Usage:
```js
const generator = new SDFGeometryGenerator( renderer );

const geometry = generator.generate( res, func, bounds );
```
Since the generator relies on the renderer, materials, shaders and render targets, I would probably vote for this design in order to delimit the class from classic geometry generators which do not have such complex dependencies.

Reply:
So, if we adopt this design, we would not be extending class, but returning a BufferGeometry on generate()?
Both options sound good to me...

Reply:
Opted for second option but not sure if makes sense to extend BufferGeometry anymore if this is a generator...
Any nicer name suggestions than SDFGeometryGenerator ? 😅

Reply:
When using `SDFGeometryGenerator` I would not extend of `BufferGeometry` anymore. Consider `generate()` like a. factory method than creates geometries. So:
```js
generate(){ let geometry = new BufferGeometry() .... return geometry; }

Reply:
> Any nicer name suggestions than SDFGeometryGenerator ? 😅

Yeah, the name is descriptive but quite long. Maybe just `SDFGenerator`?

Reply:
✅
SDF generator sounds a bit misleading... Let's leave as is, cannot come up w better one 😅